src/transformers/modeling_utils.py

```diff
@@ -3574,6 +3574,14 @@ def from_pretrained(
 
             if quantization_config is None:
                 quantization_config = AwqConfig.from_dict(config.quantization_config)
+            # In case a user passes a `AwqConfig` with `do_fuse=True` for models that have
+            # a `modules_to_not_convert` attribute we need to manually set that attribute into the
+            # passed `quantization_config`
+            elif (
+                quantization_config.modules_to_not_convert is None
+                and "modules_to_not_convert" in config.quantization_config
+            ):
+                quantization_config.modules_to_not_convert = config.quantization_config["modules_to_not_convert"]
```

Comment:
that seems a bit odd to me, should either be done in the integration (I know you don't have access to the config) or when you init the `quantization_config`, you should use `config.quantization_config` no? (at some point merging kwargs?)

Reply:
I think we always rely on `quantization_config`, we do merge kwargs but on the other way around (from `quantization_config` to `config.quantization_config`) with `get_loading_attributes()`. The scenario above happens only with the specific case where users pass `do_fuse=True` & a non-`None` value in `config.quantization_config["modules_to_not_convert"]`. I think it is a good idea to think of a way to harmonize how to merge kwargs between `config.quantization_config` and `quantization_config` but might be slightly out of the scope of the PR as I need to do it for all quantization schemes we support. I propose to do that properly in a follow up PR

Reply:
Alright let's keep that in mind